pkg/host/machine_info.go

```diff
@@ -4,58 +4,66 @@
 package host
 
 import (
-	"bytes"
-	"fmt"
 	"os"
+	"path/filepath"
 	"strings"
 )
 
-func CollectMachineInfo() ([]byte, error) {
-	buf := new(bytes.Buffer)
-	for _, pair := range machineInfoFuncs {
-		pos0 := buf.Len()
-		fmt.Fprintf(buf, "[%s]\n", pair.name)
-		pos1 := buf.Len()
-		err := pair.fn(buf)
-		if err != nil {
-			if !os.IsNotExist(err) {
-				return nil, err
-			}
-		}
-		if buf.Len() == pos1 {
-			buf.Truncate(pos0)
-			continue
-		}
-		fmt.Fprintf(buf, "\n%v\n\n", strings.Repeat("-", 80))
-	}
-	return buf.Bytes(), nil
-}
-
-func CollectModulesInfo() ([]KernelModule, error) {
-	if machineModulesInfo == nil {
-		return nil, nil
-	}
-	return machineModulesInfo()
-}
-
 func CollectGlobsInfo(globs map[string]bool) (map[string][]string, error) {
 	if machineGlobsInfo == nil {
 		return nil, nil
 	}
 	return machineGlobsInfo(globs)
 }
 
-var machineInfoFuncs []machineInfoFunc
-var machineModulesInfo func() ([]KernelModule, error)
 var machineGlobsInfo func(map[string]bool) (map[string][]string, error)
 
-type machineInfoFunc struct {
-	name string
-	fn   func(*bytes.Buffer) error
-}
-
 type KernelModule struct {
 	Name string `json:"Name"`
 	Addr uint64 `json:"Addr"`
 	Size uint64 `json:"Size"`
 }
+
+type FileInfo struct {
+	Name   string
+	Exists bool
+	Error  string
+	Data   []byte
+}
+
+func ReadFiles(files []string) []FileInfo {
+	var res []FileInfo
+	for _, glob := range files {
+		glob = filepath.FromSlash(glob)
+		if !strings.Contains(glob, "*") {
+			res = append(res, readFile(glob))
+			continue
+		}
```

Comment:
Can we remove this part and process single files as globs too?

Reply:
The reason to handle it separately was errors handing -- if the file does not exist, glob will produce 0 matches, so we will append nothing at all; if we handle it as a single file, then append an entry with an error. In the manager we print all errors we got.

Reply:
Thanks!